### PR TITLE
20190111 rup weights

### PR DESCRIPTION
**Pull request by mmpagani**
This PR improves the handling of the mutually exclusive ruptures case. Mutually exclusive ruptures are now admitted only in case of non parametric sources.
In the .xml weights are specified in this way:
``` xml
            <nonParametricSeismicSource
            id="1"
            name="Fake Non Parametric Source"
            tectonicRegion="Some TRT"
            rup_weights="0.2 0.8"
            >
```
This functionality is useful for the implementation of the New Madrid cluster case in the US model.

### CODE REVIEW

**Comment by mmpagani** [comment]
@micheles I fixed the doubled test.

File: openquake/hazardlib/tests/sourceconverter_test.py [hunk 1]
```diff
@@ -169,15 +170,33 @@ def test_complex(self):
...
                       'Active Shallow Crust, line 67', str(ctx.exception))
+
+    @raises(Exception)
+    def test_wrong_source_type(self):
+        """ Test that only nonparametric sources are used with mutex ruptures 
+        """
+        testfile = os.path.join(testdir, 'rupture_mutex_wrong.xml')
+        with self.assertRaises(ValueError) as ctx:
+            nrml.to_python(testfile)
+
```
[the remark this answers is not in the document]
Comment (mmpagani) [reply openquake/hazardlib/tests/sourceconverter_test.py]: Fixed

File: openquake/hazardlib/sourceconverter.py [hunk 2]
```diff
@@ -124,6 +131,14 @@ def _check_init_variables(self, src_list, name,
             for src in src_list:
                 assert src.tectonic_region_type == self.trt, (
                     src.tectonic_region_type, self.trt)
+                # Mutually exclusive ruptures can only belong to non-parametric
+                # sources
+                if rup_interdep in ('mutex'):
+                    print('muuutex')
```
Comment (micheles) [inline openquake/hazardlib/sourceconverter.py]: This print is a leftover

File: openquake/hazardlib/sourceconverter.py [hunk 3]
```diff
@@ -138,6 +153,14 @@ def update(self, src):
             src.tectonic_region_type, self.trt)
         if not src.min_mag:  # if not set already
             src.min_mag = self.min_mag.get(self.trt) or self.min_mag['default']
+        # checking mutex ruptures
+        if (not isinstance(src, NonParametricSeismicSource) and 
+                self.rup_interdep in ('mutex')):
```
Comment (micheles) [inline openquake/hazardlib/sourceconverter.py]: Better `self.rup_interdep == 'mutex'`

File: openquake/hazardlib/sourceconverter.py [hunk 4]
```diff
@@ -750,10 +773,15 @@ def convert_nonParametricSeismicSource(self, node):
         """
         trt = node.attrib.get('tectonicRegion')
         rup_pmf_data = []
-        for rupnode in node:
+        rups_weights = None
+        if 'rup_weights' in node.attrib.keys():
```
Comment (micheles) [inline openquake/hazardlib/sourceconverter.py]: `'rup_weights' in node.attrib` is enough

**Comment by micheles** [comment]
LGTM apart from the minor suggestions.

**Pull request merged**